include/login_functions.php

```diff
@@ -32,70 +35,88 @@ function setRememberMeCookie($userId)
     }
 
     // Update the user key
-    if (pdo_query('UPDATE ' . qid('user') . " SET cookiekey='" . $key . "' WHERE id=" . qnum($userId)) !== false) {
+    $user = new User();
+    $user->Id = $userId;
+    if ($user->SetCookieKey($key)) {
         setcookie($cookiename, $userId . $key, $time);
     }
 }
 
 /** Database authentication */
 function databaseAuthenticate($email, $password, $SessionCachePolicy, $rememberme)
 {
-    global $loginerror;
+    global $CDASH_COOKIE_EXPIRATION_TIME, $CDASH_EXTERNAL_AUTH, $loginerror;
     $loginerror = '';
 
-    include dirname(__DIR__) . '/config/config.php';
-
-    $db = pdo_connect("$CDASH_DB_HOST", "$CDASH_DB_LOGIN", "$CDASH_DB_PASS");
-    pdo_select_db("$CDASH_DB_NAME", $db);
-    $sql = 'SELECT id,password FROM ' . qid('user') . " WHERE email='" . pdo_real_escape_string($email) . "'";
-    $result = pdo_query("$sql");
-
-    if (pdo_num_rows($result) == 0) {
-        pdo_free_result($result);
+    $user = new User();
+    $userid = $user->GetIdFromEmail($email);
+    if (!$userid) {
         $loginerror = 'Wrong email or password.';
         return false;
     }
 
-    $user_array = pdo_fetch_array($result);
-    $pass = $user_array['password'];
+    $user->Id = $userid;
+    $user->Fill();
 
     // Check if the account is locked out.
-    if (accountIsLocked($user_array['id'])) {
+    if (accountIsLocked($userid)) {
         return false;
     }
 
-    // External authentication
     if ($password === null && isset($CDASH_EXTERNAL_AUTH) && $CDASH_EXTERNAL_AUTH) {
-        // create the session array
-        $sessionArray = array('login' => $login, 'password' => 'this is not a valid password', 'passwd' => $user_array['password'], 'ID' => session_id(), 'valid' => 1, 'loginid' => $user_array['id']);
+        // External authentication succeeded.
+        // Create the session array.
+        $sessionArray = array('login' => $login, 'password' => 'this is not a valid password', 'passwd' => $row['password'], 'ID' => session_id(), 'valid' => 1, 'loginid' => $userid);
```

Comment:
You're right, good catch.  Related to your other comment, maybe we don't need to have this info in the session anyway.